### ISSUE TITLE
zerfoo#872 loss-seed onesLike breaks CUDA-graph capture (host cudaMemcpy on legacy stream during capture)

### ISSUE THREAD

**Issue by dndungu** [zerfoo/zerfoo#875]
## Summary

The zerfoo#872 fix (v1.50.1) seeds `loss.Backward` with a **freshly host-allocated** ones tensor (`onesLike`) on every step. Inside a CUDA-graph capture region this triggers a host->device `cudaMemcpy` on the legacy stream, which CUDA forbids during stream capture, so **all capture-on GPU training now fails**:

```
loss backward pass failed: cudaMemcpy failed: operation would make the
legacy stream depend on a capturing blocking stream
```

This is a regression introduced by #872 — capture-replay (the `CaptureReplayRunner`, #869) worked before the loss seed flipped from the (device-resident) loss tensor to a host-allocated ones tensor.

## Root cause

`training/strategy_common.go` `computeGradientsTensorCommon`:

```go
ones, err := onesLike[T](g.Engine(), lossTensor)   // host-backed (CPUStorage) tensor, allocated per step
...
lossGrads, err := loss.Backward(ctx, mode, ones, output, batch.Targets)
```

`onesLike` returns a plain host (CPUStorage) tensor. The loss layer's `Backward` ends with `engine.Mul(localGrad, dOut)`, so `dOut` (the host `ones`) must be uploaded to the device — `getDevicePtr` issues a `cudaMemcpy` on the default/legacy stream. During `BeginCapture/EndCapture` that is an illegal stream dependency, and the captured walk aborts.

Confirmed via `ZERFOO_ARENA_PROFILE=1`: the **only** `getDevicePtr` arena upload in the entire captured walk is a 4-byte scalar (the `[1]` ones seed), immediately before the `cudaMemcpy` failure.

## Reproduction

- zerfoo v1.50.1, ztensor v1.11.1, GB10 (arm64 sm_121), f32 CrossEntropyLossOneHot.
- Any `CaptureReplayRunner`-driven training step (warmup eager step OK; the **capture** step fails).
- Wolf CrossAsset GPU training (both per-sample `[ns,3]` and batched `[B*ns,3]`) reproduces identically.

## Suggested fix (general)

Make the loss-backward seed device-resident and allocated **outside** the capture region:

- Allocate the ones seed once per loss node (engine-owned device buffer, sized to the loss tensor shape) and reuse it across steps — like the ztensor PR #145 per-signature device-cache pattern for kernel parameter arrays; or
- Have `onesLike` return an engine/arena-resident tensor (already on-device) when `g.Engine()` is a GPU engine, so `engine.Mul` needs no per-step H2D copy; or
- Seed via an engine op (`engine.Fill`/`FillScalar` into a stable device buffer) rather than a host-allocated tensor.

The seed is shape `[1]` for a scalar loss (broadcasts), so a single cached device scalar suffices for the common case.

## Impact

Blocks **all** capture-on GPU training (the only path that removes per-launch overhead). Wolf's E13 batch-vectorization parity bench can only measure **eager** throughput until this lands; capture-on (the ~2x per-launch win) is unavailable. This is general (not Wolf-specific): any capture-on consumer of `DefaultBackpropStrategy` hits it.
